web-common/src/features/dashboards/filters/dimension-filters/dimension-filter-values.ts

```diff
@@ -41,19 +46,29 @@ export function useDimensionSearch(
     timeStart,
     timeEnd,
     enabled,
-    additionalFilter,
+
+    metricsViewWheres,
   }: DimensionSearchArgs,
 ) {
-  const where = getFilterForSearchArgs(dimensionName, {
-    mode,
-    searchText,
-    values,
-    additionalFilter,
-  });
-
   // Main query: Get top 250 results (above the fold)
-  const mainQueries = metricsViewNames.map((mvName) =>
-    createQueryServiceMetricsViewAggregation(
+  const mainQueries = metricsViewNames.map((mvName) => {
+    const where = getFilterForSearchArgs(dimensionName, {
+      mode,
+      searchText,
+      values,
+      additionalFilter: sanitiseExpression(
+        mergeDimensionAndMeasureFilters(
```

Comment:
Call to `mergeDimensionAndMeasureFilters` is redundant if we are passing empty array in this file.